babel/numbers.py

```diff
@@ -641,68 +654,134 @@ def __init__(self, pattern, prefix, suffix, grouping,
         self.frac_prec = frac_prec
         self.exp_prec = exp_prec
         self.exp_plus = exp_plus
-        if '%' in ''.join(self.prefix + self.suffix):
-            self.scale = 2
-        elif u'‰' in ''.join(self.prefix + self.suffix):
-            self.scale = 3
-        else:
-            self.scale = 0
 
     def __repr__(self):
         return '<%s %r>' % (type(self).__name__, self.pattern)
 
-    def apply(self, value, locale, currency=None, force_frac=None):
-        frac_prec = force_frac or self.frac_prec
+    @property
+    def scale(self):
+        """Scaling factor to apply to the number before rendering.
+
+        Auto-set to a factor of 2 or 3 if presence of a % or ‰ sign is
+        detected in the pattern. Default is to not mess with the scale at all.
+        """
+        scale = 0
+        if '%' in ''.join(self.prefix + self.suffix):
+            scale = 2
+        elif u'‰' in ''.join(self.prefix + self.suffix):
```

Comment:
The call to `join` is indeed required as `prefix` and `suffix` are lists of strings. Removing the `join` introduce a regression. See: https://travis-ci.org/python-babel/babel/jobs/240689769#L1007

I reverted the change in 4887fdfa695a93b7d148057c167792f3a98a8912. All unittests passes now.